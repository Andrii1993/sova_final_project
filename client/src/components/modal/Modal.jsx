import { ModalContainer,ModalHeader,ModalClose,ModalFooter,ModalMain,OutsideModal } from './styles';
import { ReactComponent as MyLogo }  from './cross.svg'
import PropTypes from 'prop-types';
const Modal = ({header,text,closeButton,onClose,children}) => {
    const btnClose = closeButton?<ModalClose><MyLogo onClick={onClose} style={{width: "inherit",height: "inherit"}}/></ModalClose>:null;
    return (
      <>
    <OutsideModal onClick={onClose}/>
    <ModalContainer className='modal'>
     <ModalHeader>
      <h1>{header}</h1>
      {btnClose}
     </ModalHeader>
     <ModalMain>
      <p>{text}</p>
     </ModalMain>
     <ModalFooter>
      {children}
     </ModalFooter>
    </ModalContainer>
    </>
      )
}

Modal.propTypes = {
  header: PropTypes.string,
  text: PropTypes.string,
  onClose: PropTypes.func
};

Modal.defaultProps = {
  header: "Message",
  text: "",
  onClose: null
};

export default Modal;
