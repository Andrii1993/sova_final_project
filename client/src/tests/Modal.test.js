import Modal from '../components/modal/Modal';

import {cleanup,render} from "@testing-library/react";
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders with a header text", () => {
  const{container}= render(<Modal header="header"/>);
  expect(container.textContent).toBe("header");
});

it("renders with a  text", () => {
  const{container}=render(<Modal header="modal" text="text"/> );
  expect(container.textContent).toBe("modaltext");
});

it('test SnapShot', () => {
  const modalComp = renderer.create(<Modal header="header" text="text"><button/></Modal>).toJSON()
  expect(modalComp).toMatchSnapshot(); 
});



