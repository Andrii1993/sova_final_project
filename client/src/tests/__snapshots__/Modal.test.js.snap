// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SnapShot 1`] = `
Array [
  <div
    className="sc-bczRLJ duYrqX"
    onClick={null}
  />,
  <div
    className="sc-gsnTZi giEWGm modal"
  >
    <div
      className="sc-hKMtZM cDhhft"
    >
      <h1>
        header
      </h1>
    </div>
    <div
      className="sc-eCYdqJ frHuIF"
    >
      <p>
        text
      </p>
    </div>
    <div
      className="sc-jSMfEi ivhTau"
    >
      <button />
    </div>
  </div>,
]
`;
